generator/generator.js

```diff
@@ -18,53 +24,159 @@ program
   .addOption(
-    new Option(
-      "-name, --name <string>",
-      "name of the proposal (e.g. CapsIncrease)"
-    ).makeOptionMandatory()
-  )
-  .addOption(
-    new Option("-ch, --chains <letters...>")
-      .choices([
-        "Ethereum",
-        "Optimism",
-        "Arbitrum",
-        "Polygon",
-        "Avalanche",
-        "Fantom",
-        "Harmony",
-        "Metis",
-      ])
-      .makeOptionMandatory()
-  )
-  .addOption(
-    new Option("-pv, --protocolVersion <string>")
-      .choices(["V2", "V3"])
-      .makeOptionMandatory()
+    new Option("-pv, --protocolVersion <string>").choices(AVAILABLE_VERSIONS)
   )
   .addOption(new Option("-t, --title <string>", "aip title"))
   .addOption(new Option("-a, --author <string>", "author"))
   .addOption(new Option("-d, --discussion <string>", "forum link"))
   .addOption(new Option("-s, --snapshot <string>", "snapshot link"))
-  .allowExcessArguments(false);
-
-program.parse();
+  .allowExcessArguments(false)
+  .parse(process.argv);
 
 const options = program.opts();
 
-const baseName = generateName(options);
+// workaround as there's validate is not currently supported on checkbox
+// https://github.com/SBoudrias/Inquirer.js/issues/1257
+while (!options.chains?.length === true) {
+  options.chains = await checkbox({
+    message: "Chains this proposal targets",
+    choices: AVAILABLE_CHAINS.map((v) => ({ name: v, value: v })),
+    validate(input) {
+      // currently ignored due to a bug
+      if (input.length == 0)
+        return "You must target at least one chain in your proposal!";
+      return true;
+    },
+  });
+}
 
-// create files
+if (!options.protocolVersion) {
+  options.protocolVersion = await select({
+    message: "Protocol version this proposal targets",
+    choices: AVAILABLE_VERSIONS.map((v) => ({ name: v, value: v })).reverse(),
+    default: "V3", // default on select not currently supported which is why we reverse
+  });
+}
+
+/**
+ * TODO: config engine flag is a bit arbitrary.
+ * Would be better to ask for specific features & inline proper tests and boilerplate
+ */
+if (!options.configEngine) {
+  options.configEngine = await confirm({
+    message: "To you plan to use the config engine?",
+    default: true,
+  });
+}
+
+export const ENGINE_FEATURES = {
+  preHook: {
+    name: "PreHook",
+    value: "PreHook",
+  },
+  postHook: {
+    name: "PostHook",
+    value: "PostHook",
+  },
+  rateStrategiesUpdates: {
+    name: "RateStrategyUpdates",
+    value: "RateStrategyUpdates",
+  },
+  capsUpdate: {
+    name: "CapsUpdate",
+    value: "CapsUpdate",
+  },
+};
+
+if (options.configEngine) {
+  if (options.protocolVersion === "V2") {
+    options.features = await checkbox({
+      message: "What do you want to do?",
+      choices: [
+        ENGINE_FEATURES.rateStrategiesUpdates,
+        ENGINE_FEATURES.preHook,
+        ENGINE_FEATURES.postHook,
+      ],
+    });
+
+    // TODO: collect more info regarding certain features
```

Comment:
Which features are pending here? Are we aiming to support pre-built generation for all possible updates (like changing an oracle for example)?

Reply:
Over time i think it makes sense to have a engine to build all kind of proposals.
Highly depends on @marczeller etc willigness to use/improve it as in the end it's a tool from/by contributors.

Reply:
So for now we're good with leaving this comment and keeping it open for external contributions yeah?

Reply:
i'd say so, i mean we can also improve it, but if we want to cover 100% we'll never release it and it's not enough priority to actively work on it right now.